src/mame/itech/itech8_m.cpp

```diff
@@ -306,53 +300,53 @@ void slikshot_state::vels_to_inters(u8 x, u8 vx, u8 vy,
  *
  *************************************/
 
-void slikshot_state::inters_to_words(u16 inter1, u16 inter2, u16 inter3, u8 *beams,
-							u16 *word1, u16 *word2, u16 *word3)
+void slikshot_state::inters_to_words(u16 inter1, u16 inter2, u16 inter3, u8 &beams,
+							u16 &word1, u16 &word2, u16 &word3)
 {
 	u16 word2mod;
 
-	/* intermediate value 3 is always equal to the third word */
-	*word3 = inter3;
+	// intermediate value 3 is always equal to the third word
+	word3 = inter3;
 
-	/* on input, it is expected that the low bit of beams has already been determined */
-	if (*beams & 1)
+	// on input, it is expected that the low bit of beams has already been determined
+	if (beams & 1)
 	{
-		/* make sure we can do it */
+		// make sure we can do it
 		if (inter3 <= inter1)
 		{
-			/* always go back via case 3 */
-			*beams |= 2;
+			// always go back via case 3
+			beams |= 2;
 
-			/* compute an appropriate value for the scaled version of word 2 */
+			// compute an appropriate value for the scaled version of word 2
 			word2mod = inter1 - inter3;
 
-			/* compute the other values from that */
-			*word1 = inter2 - word2mod;
-			*word2 = ((uint64_t)word2mod << 16) / 0x16553;
+			// compute the other values from that
+			word1 = inter2 - word2mod;
+			word2 = ((u64)word2mod << 16) / 0x16553;
 		}
 		else
 			LOGSENSOR("inters_to_words: unable to convert %04x %04x %04x %02x\n",
-					(u32)inter1, (u32)inter2, (u32)inter3, (u32)*beams);
+					(u32)inter1, (u32)inter2, (u32)inter3, (u32)beams);
```

Comment:
Why the casts to `u32` here?  The logging functions work fine with any width integer types.